finetune/base_models/oscar/featurizer.py

```diff
@@ -0,0 +1,261 @@
+import tensorflow as tf
+from finetune.base_models.gpt.featurizer import dropout, embed, split_heads, merge_heads
+from finetune.util.shapes import shape_list
+from finetune.base_models.oscar.ra import recursive_agg
+from finetune.optimizers.recompute_grads import recompute_grad
+from finetune.nn.add_auxiliary import add_auxiliary
+
+import functools
+
+
+def cast_maybe(t, dtype):
+    if t.dtype == dtype:
+        return t
+    return tf.cast(t, dtype=dtype)
```

Comment:
Oh yikes -- does tf put the op in the graph even if it's already of the given dtype?